WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift

```diff
@@ -64,94 +75,102 @@ public class NotificationSettingDetailsViewController : UITableViewController
+        let row             = Row(kind: .Text, description: description, key: nil, value: nil)
+        
+        let footerText      = NSLocalizedString("Push Notifications have been turned off in iOS Settings App. " +
+                                                "Toggle \"Allow Notifications\" to turn them back on.",
+                                                comment: "Suggests to enable Push Notification Settings in Settings.app")
+        let section         = Section(rows: [row], footerText: footerText)
+        
+        return [section]
     }
     
 
     
     // MARK: - UITableView Delegate Methods
     public override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
-        return sectionCount
+        return sections.count
     }
     
     public override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return isDeviceStreamDisabled() ? disabledRowCount : rows.count
+        return sections[section].rows.count
     }
     
     public override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        // Disabled Push Notifications:
-        // We'll render just one row, that should open Settings.app on press
-        //
-        if isDeviceStreamDisabled() {
-            let cell                = tableView.dequeueReusableCellWithIdentifier(defaultIdentifier) as! WPTableViewCell
-            cell.textLabel?.text    = NSLocalizedString("Go to iPhone Settings", comment: "Opens WPiOS Settings.app Section")
-            WPStyleGuide.configureTableViewCell(cell)
-            
-            return cell
-        }
+        let section = sections[indexPath.section]
+        let row     = section.rows[indexPath.row]
+        let cell    = tableView.dequeueReusableCellWithIdentifier(row.kind.rawValue) as! UITableViewCell
         
-        // Settings:
-        // One SwitchCell per setting!
-        //
-        let cell = tableView.dequeueReusableCellWithIdentifier(switchIdentifier) as! SwitchTableViewCell
-        configureSwitchCell(cell, indexPath: indexPath)
+        switch row.kind {
+        case .Text:
+            configureTextCell(cell as! WPTableViewCell, row: row)
+        case .Setting:
+            configureSwitchCell(cell as! SwitchTableViewCell, row: row)
+        }
         
         return cell
     }
     
     public override func tableView(tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
-        if !isDeviceStreamDisabled() {
-            return CGFloat.min
+        if let footerText = sections[section].footerText {
+            return WPTableViewSectionHeaderFooterView.heightForFooter(footerText, width: view.bounds.width)
         }
-        
-        return WPTableViewSectionHeaderFooterView.heightForFooter(disabledDeviceFooterText(), width: view.bounds.width)
+
+        return CGFloat.min
```

Comment:
`CGFloat.min` usage, in this spot, would be [because of this](http://stackoverflow.com/questions/27321237/uitableview-headerview-different-sizes). I'm afraid that simply returning 0 causes the footer to assume a default height (weird right!?).

Sample shots of `0` vs `.min` [here here](https://cloudup.com/cbyB59zIpXo).

Not sure precisely what happens within the `UITableView`'s realm itself, since that fellow expects a CGFloat, and rounding per se is up to the caller's code.

Thanks!